### PR TITLE
Add UTF32Encoding tests

### PR DESCRIPTION
**Pull request by hughbe**
- Encoding and decoding valid chars and bytes
- Other methods
- Ctors
- Properties

/cc @tarekgh

### CODE REVIEW

File: src/System.Text.Encoding/tests/UTF32Encoding/UTF32EncodingTests.cs [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+            yield return new object[] { new UTF32Encoding(), new object(), false };
+            yield return new object[] { new UTF32Encoding(), null, false };
+        }
+
+        [Theory]
+        [MemberData(nameof(Equals_TestData))]
+        public void Equals(UTF32Encoding encoding, object value, bool expected)
+        {
+            Assert.Equal(expected, encoding.Equals(value));
+            if (value is UTF32Encoding)
```
Comment (tarekgh) [inline src/System.Text.Encoding/tests/UTF32Encoding/UTF32EncodingTests.cs]: Look at my comment on the PR https://github.com/dotnet/corefx/pull/8021/files
what you think?

Comment (hughbe) [inline src/System.Text.Encoding/tests/UTF32Encoding/UTF32EncodingTests.cs]: this can be simplified. fixed

**Comment by tarekgh** [comment]
LGTM. thanks.

**Comment by tarekgh** [comment]
Test Innerloop OSX Debug Build and Test please

**Comment by tarekgh** [comment]
@hughbe could you resolve the conflicts?

**Comment by hughbe** [comment]
Whoops I didn't realise. Will do

**Comment by hughbe** [comment]
Fixed merge conflict

**Pull request merged**